addons/account_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,307 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields, api
+from odoo.exceptions import UserError
+from odoo.tools.pdf import OdooPdfFileReader, OdooPdfFileWriter
+
+from lxml import etree
+import base64
+import io
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountEdiFormat(models.Model):
+    _name = 'account.edi.format'
+    _description = 'EDI format'
+
+    name = fields.Char(index=True)
+    code = fields.Char()
+    hide_on_journal = fields.Boolean('Is exported supported for this EDI ?')
+
+    _sql_constraints = [
+        ('unique_code', 'unique (code)', 'This code already exists')
+    ]
+
+    ####################################################
+    # Export method to override based on EDI Format
+    ####################################################
+
+    def _export_invoice_to_attachment(self, invoice):
+        """ Create the file content representing the invoice.
+
+        :param invoice: the invoice to encode.
+        :returns: a dictionary (values are compatible to create an ir.attachment)
+        * name : the name of the file
+        * datas : the content of the file,
+        * res_model : 'account.move',
+        * res_id: the id of invoice
+        * mimetype : the mimetype of the attachment
+        """
+        # TO OVERRIDE
+        self.ensure_one()
+        return False
+
+    def _export_invoice_to_embed_to_pdf(self, pdf_content, invoice):
+        """ Create the file content representing the invoice to embed into pdf.
+            This hook allows to customize the EDI format generation based on the pdf.
+            For example : to include the pdf inside the EDI format.
+            The default behavior is to call _export_invoice_to_attachment.
+            Should return False for EDI format that must not be embedded to pdf. 
+
+            :param pdf_content: the pdf before any EDI format was added.
+            :param invoice: the invoice to add.
+            :returns: a dictionary or False if this EDI format must not be embedded to pdf.
+            * name : the name of the file
+            * datas : the content of the file,
+            * res_model : 'account.move',
+            * res_id: the id of invoice
+            * mimetype : the mimetype of the attachment
+        """
+        # TO OVERRIDE
+        self.ensure_one()
+        return self._export_invoice_to_attachment(invoice)
```

Comment:
ho I get i! I was suggesting to return False, by default.
 
But no, what we want is
* the default is to add the EDI document inside the PDF, because we find it convenient 
* but one may want not to do it (and return False instead)
* but one may need to do some special treatment before and overwrite the behavior

Was it clear enough from the docstring?